Tools/gittools/pre-commit.py

```diff
@@ -15,27 +15,35 @@
 import subprocess
 
 
-class PreCommitFlake8(object):
+class AP_PreCommit(object):
 
     def __init__(self):
         pass
 
-    def progress(self, message):
+    @staticmethod
+    def progress(message):
         print("***** %s" % (message, ))
 
-    def check_file(self, filepath):
+    @staticmethod
+    def has_flake8_tag(filepath):
         content = open(filepath).read()
-        if "AP_FLAKE8_CLEAN" not in content:
+        if "AP_FLAKE8_CLEAN" in content:
             return True
-        self.progress("Checking (%s)" % filepath)
-        retcode = subprocess.call(["flake8", filepath])
-        if retcode != 0:
-            self.progress("File (%s) failed with retcode (%s)" %
-                          (filepath, retcode))
-            return False
-        return True
+        return False
 
-    def split_git_diff_output(self, output):
+    def check_flake8(self, files_to_check):
+        if files_to_check:
+            for path in files_to_check:
+                self.progress("Checking (%s)" % path)
+            ret = subprocess.run(["flake8"] + files_to_check,
+                                 stdout=subprocess.PIPE, stderr=subprocess.STDOUT, text=True)
+            if ret.returncode != 0:
+                self.progress("Flake8 check failed: (%s)" % (ret.stdout))
+                return 1
```

Comment:
These pytthon methods should be returning true or false, not a retcode.

Reply:
Fixed. I think the return 1/0 at the end of run() is still needed. Clarity maybe in that case?